### DISCUSSION TITLE
How to disable saving transcript results? (CLI)

### DISCUSSION THREAD

**Discussion by hustLer2k** [openai/whisper#1890] [Q&A] [answered]
I want to just pipe transcription to another process without transcription results being saved.

**Comment by ryanheise** [comment, marked as answer]
You can't do that with the CLI (there is no option for it).

The easiest workaround would be to run whisper, then `cat` the file, piping that into another process, and removing the file afterwards.

A more involved way would be to write your own Python script, since then you could invoke `result = transcribe("large-v2", "audio.mp3")` method and then print it to stdout with `print(json.dumps(result))` for example. That way no file will be written and the transcript will be printed to stdout. If you want to print out just the text, you can select that from the json before printing. Unfortunately you can't use any of the SRT/VTT/etc writers since those are all hard coded to write to files.

**Reply by hustLer2k** [reply]
Thanks. Is there a way to output the data in chunks from the script (to implement streaming in our system) instead waiting for the full operation to complete?

**Reply by ryanheise** [reply]
In Python, you could use `transcribe("large-v2", "audio.mp3", "verbose=True")` and it should print out the same progressive output to stdout that you see when using the CLI, but without saving the output to a file. However, since `verbose=True` is intended only as a progress indicator, there is no flexibility to control the format of this output and there is a chance that the format of this verbose output could change in future versions. If that's satisfactory, you could pin your app to a specific version of Whisper and check before upgrading whether it changes the format. If not, you could fork whisper and see where in `transcribe.py` it prints out that verbose output and you could modify that line of code to print it out in a different format.
